contracts/proxies/SafeProxy.sol

```diff
@@ -33,10 +33,10 @@ contract SafeProxy {
     fallback() external payable {
         // solhint-disable-next-line no-inline-assembly
         assembly {
-            let _singleton := and(sload(0), 0xffffffffffffffffffffffffffffffffffffffff)
+            let _singleton := sload(0)
             // 0xa619486e == keccak("masterCopy()"). The value is right padded to 32-bytes with 0s
             if eq(calldataload(0), 0xa619486e00000000000000000000000000000000000000000000000000000000) {
-                mstore(0, _singleton)
+                mstore(0, and(_singleton, 0xffffffffffffffffffffffffffffffffffffffff))
```

Comment:
We might be able to save bytecode here by masking with shifting instead of `and`-ing, which would decrease deployment costs.

```suggestion
                mstore(12, shl(12, _singleton))
```

Would be good to quantify codesize before and after to see if its worth it (200 gas per deployed byte).

Reply:
We could even do something like this:

```suggestion
                mstore(12, shl(12, _singleton))
```

Which should have equivalent gas costs, but I don't know if memory `0-12` is guaranteed to be 0'ed out...

Reply:
I pushed a commit applying the first, the later will further reduce 3 bytes:

SafeProxy 152 bytes (limit is 24576)